python/dlisio/ext/core.cpp

```diff
@@ -698,13 +700,33 @@ noexcept (false) {
     assert(std::string(pre_fmt) == "");
     assert(std::string(post_fmt) == "");
 
+    auto record_dst = dst;
+
+    const auto handle = [&]( const std::string& problem ) {
+        const auto context = "curves (read_fdata)";
+        errorhandler.log(dl::error_severity::CRITICAL, context, problem, "",
+                         "Record is skipped");
+        // we update the buffer as we go. Hence if error happened we need to
+        // go back and start rewriting updated data
+        dst = record_dst;
+    };
+
     int frames = 0;
     for (auto i : indices) {
+        record_dst = dst;
+
         /* get record */
-        auto record = dl::extract(file, i);
+        dl::record record;
+        try {
+            record = dl::extract(file, i);
+        } catch (std::exception& e) {
+            handle(e.what());
+            continue;
```

Comment:
At the start of read_fdata we allocate rows equal to the number of fdata. What happens to the array when we don't fill it up completely because we skip some frames? Should it be resized?

Reply:
You are completely right, my bad